gw_eccentricity/eccDefinition.py

```diff
@@ -192,12 +234,15 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         if not np.allclose(self.t_diff, self.t_diff[0]):
             raise Exception("Input time array must have uniform time steps.\n"
                             f"Time steps are {self.t_diff}")
-        self.hlm = self.dataDict["hlm"]
-        self.h22 = self.hlm[(2, 2)]
-        self.amp22 = np.abs(self.h22)
-        self.phase22 = - np.unwrap(np.angle(self.h22))
-        self.omega22 = time_deriv_4thOrder(self.phase22,
-                                           self.t[1] - self.t[0])
+        # get phase22 from dataDict
+        self.phase22, self.omega22, self.h22 = self.compute_phase22(self.dataDict)
+        # get amp22 from dataDict
+        self.amp22, _ = self.compute_amp22(self.dataDict)
+        # get omega22 from dataDict
+        # check if omega22 has been already computed when getting
+        # phase22.
+        if self.omega22 is None:
+            self.omega22 = self.compute_omega22(self.dataDict)
```

Comment:
Just do this in that one function?

Reply:
Do you mean putting these lines for getting amp22, phase22, and omega22 in a single wrapper function and calling it? As mentioned above, I don't think writing a single function that gets all these is better than keeping separate functions for each of these.